LanguageFeatures/Augmentation-libraries/augmented_expression_A02_t18.dart

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The exact result of an `augmented` expression depends on what is
+/// being augmented, but it generally follows the same rules as any normal
+/// identifier:
+/// ...
+/// - Augmenting setters: Within an augmenting setter `augmented` must be
+///   followed by an `=` and will directly invoke the augmented setter. If
+///   augmenting a field with a setter, this will invoke the implicit setter
+///   from the augmented field.
+///
+/// @description Checks that it is a compile-time error for an augmenting
+/// setter to have as a parameter of a type with the name `augmented`.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+import augment 'augmented_expression_A02_t18_lib.dart';
+
+class augmented {}
+
+void set topLevelSetter(String value) {}
```

Comment:
Probably:
```suggestion
void set topLevelSetter(augmented value) {}
```
Similarly for 8 locations below.

Reply:
No, we tested it in a previous test (`...A02_t17.dart`). Here we are testing use a class  `augmented` in a body of a setter. Description updated

Reply:
Makes sense, thanks!